File: swift/sum_of_even_squares.swift
func sumOfEvenSquares(of a: [Int]) -> Int {
    var sum = 0
    for x in a where x.isMultiple(of: 2) {
        sum += x * x
    }
    return sum
}

assert(sumOfEvenSquares(of: []) == 0)
assert(sumOfEvenSquares(of: [1]) == 0)
assert(sumOfEvenSquares(of: [1,2]) == 4)
assert(sumOfEvenSquares(of: [1,2,3]) == 4)
assert(sumOfEvenSquares(of: [1,2,3,4]) == 20)

File: python/sum_of_even_squares.py
def sum_of_even_squares(a):
    return sum(x*x for x in a if x % 2 == 0)

assert sum_of_even_squares([]) == 0
assert sum_of_even_squares([1]) == 0
assert sum_of_even_squares([1,2]) == 4
assert sum_of_even_squares([1,2,3]) == 4
assert sum_of_even_squares([1,2,3,4]) == 20
